extensions/wikia/UserLogin/UserLoginSpecialController.class.php

```diff
@@ -257,72 +257,87 @@ public function getCloseAccountRedirectUrl() {
 	}
 
 	/**
-	 * @brief renders html version that will be inserted into ajax based login interaction
-	 * @details
-	 *   on GET, template partial for an ajax element will render
+	 * Renders html version that will be inserted into ajax based login interaction.
+	 * On GET, template partial for an ajax element will render
 	 */
 	public function dropdown() {
-		$query = $this->app->wg->Request->getValues();
-
-		if ( isset( $query['password'] ) ) {
-			// remove the password from the params to prevent exposiong in into the URL
-			unset($query['password']);
-		}
+		$query = $this->getSanitizedQueryString();
 
-		$this->returnto = $this->getReturnToFromQuery( $query );
-		$this->returntoquery = $this->getReturnToQueryFromQuery( $query );
+		$this->response->setData( [
+			'returnto' => $this->getReturnToFromQuery( $query ),
+			'returntoquery' => $this->getReturnToQueryFromQuery( $query ),
+		] );
 
 		$requestParams = $this->getRequest()->getParams();
 		if ( !empty( $requestParams[ 'registerLink' ] ) ) {
-			$this->registerLink = $requestParams[ 'registerLink' ];
+			$this->response->setVal( 'registerLink',  $requestParams[ 'registerLink' ] );
 		}
+
 		if ( !empty( $requestParams[ 'template' ] ) ) {
 			$this->overrideTemplate( $requestParams[ 'template' ] );
 		}
 	}
 
-	public function getMainPagePartialUrl() {
-		return Title::newMainPage()->getPartialURL();
-	}
-
 	/**
-	 * @param String $title
-	 *
-	 * @return Boolean
+	 * This method returns an array of query parameters that do not contain sensitive information
+	 * This can be used to construct returnto and returntoquery values that are not compromising.
 	 */
-	public function isTitleBlacklisted( $title ) {
-		return AccountNavigationController::isBlacklisted( $title );
+	private function getSanitizedQueryString() {
+		$query = $this->app->wg->Request->getValues();
+		$filterParams = [ 'password', 'username', 'loginToken' ];
```

Comment:
We may want to add `userloginext01` and `userloginext02` to this list since those are synonyms with `username` and `password` at this point. In my work for SOC-163 I'm replacing each instance of these strings with a constant, but maybe just use the strings for now.